### ISSUE TITLE
#66: Train XGBoost classifier with Optuna hyperparameter optimization

### ISSUE THREAD

**Issue by ishaq2321** [ishaq2321/phishing-detection-osint#66]
## Objective
Train an XGBoost binary classifier for phishing detection with automated hyperparameter optimization using Optuna. This is the CORE ML deliverable of the thesis.

## Background
XGBoost is the gold-standard for tabular classification tasks, consistently winning Kaggle competitions and cited in 10,000+ academic papers. Combined with Optuna (Bayesian hyperparameter optimization), this produces a well-tuned, production-grade model.

## Tasks
- [ ] Create `backend/ml/training/trainModel.py`
- [ ] Create `notebooks/01_model_training.ipynb` (for thesis documentation with inline plots)
- [ ] Load `data/processed/train.csv` and `data/processed/val.csv`
- [ ] Define XGBoost training configuration:
  ```python
  base_params = {
      "objective": "binary:logistic",
      "eval_metric": ["logloss", "auc", "error"],
      "tree_method": "hist",   # Fast histogram-based
      "verbosity": 1,
      "seed": 42
  }
  ```
- [ ] Implement Optuna hyperparameter search:
  - `max_depth`: [3, 10]
  - `learning_rate`: [0.01, 0.3]
  - `n_estimators`: [100, 1000]
  - `min_child_weight`: [1, 10]
  - `subsample`: [0.6, 1.0]
  - `colsample_bytree`: [0.6, 1.0]
  - `gamma`: [0, 5]
  - `reg_alpha`: [0, 10]
  - `reg_lambda`: [0, 10]
  - 50 trials, optimize for validation AUC
- [ ] 5-fold stratified cross-validation on training set
- [ ] Early stopping: stop if validation AUC does not improve for 50 rounds
- [ ] Save best model to `backend/ml/models/phishingModel.json` (XGBoost native JSON format)
- [ ] Save Optuna study results to `data/evaluation/optuna_study.json`
- [ ] Save training history (loss curves) for plotting
- [ ] Log: best hyperparameters, best CV AUC, training time

## Acceptance Criteria
- Model achieves > 0.95 AUC on validation set (expected for phishing detection with 29 features)
- Model file saved as JSON (portable, no pickle security issues)
- Optuna study is reproducible (seeded)
- Training notebook documents the entire process with plots
- Model file size < 5MB

## Dependencies
- #65 (prepared dataset)
